tango/step.py

```diff
@@ -306,7 +306,9 @@ def _run_with_work_dir(self, workspace: "Workspace", **kwargs) -> T:
                 if self.cache_results:
                     result = workspace.step_finished(self, result)
                 return result
-            except Exception as e:
+            except BaseException as e:
+                # TODO (epwalsh): do we want to handle KeyboardInterrupts differently?
```

Comment:
I think "failed" is a perfectly acceptable state for this until we find we need to do something special.